pkg/apis/machine/v1beta1/machinehealthcheck_types.go

```diff
@@ -115,7 +115,8 @@ type MachineHealthCheckStatus struct {
 	// RemediationsAllowed is the number of further remediations allowed by this machine health check before
 	// maxUnhealthy short circuiting will be applied
 	// +kubebuilder:validation:Minimum=0
-	RemediationsAllowed int32 `json:"remediationsAllowed,omitempty"`
+	// +optional
+	RemediationsAllowed int32 `json:"remediationsAllowed"`
```

Comment:
Makes sense to me, but is this expected from optional? The docs mention that this option works only with pointers. Aren't we using some side-effect?

Reply:
I haven't seen that in the docs 🤔  As far as I'm aware you can mark any field optional, and if you don't it's assumed required if it's a reference instead of a pointer. I think this is the intended usage of this.

Without the optional marker, it is marked as required in the CRD schema and then breaks the controller if it's not been initialised yet. This approach seems to work from my testing.